src/components/UserProfileSettings/AddEmail.tsx

```diff
@@ -0,0 +1,263 @@
+    );
+  };
+
+  const resendVerificationCode = () => {
+    handleSendVerificationCode();
+  };
+
+  return (
+    <Modal
+      size="small"
+      isOpen={isOpen}
+      onClose={onClose}
+      acceptButtonProps={
+        step === Steps.EnterEmail
+          ? {
+              children: 'Next',
+              loading: isSendingVerification,
+              onClick: () => {
+                emailFormik.handleSubmit();
+              },
+            }
+          : null
+      }
+      cancelButtonProps={null}
+    >
+      {step === Steps.EnterEmail && (
+        <Box width="100%">
+          <Text
+            textAlign="center"
+            variant="h4-semibold"
+          >
+            Enter your email
+          </Text>
+
+          <Text
+            textAlign="center"
+            variant="bs-regular"
+          >
+            Confirm your email and verify to connect
+          </Text>
+
+          <form onSubmit={emailFormik.handleSubmit}>
+            <Box
+              margin="spacing-md spacing-none spacing-none spacing-none"
+              width="100%"
+            >
+              <TextInput
+                label="Email Address"
+                value={emailFormik.values.email}
+                onChange={emailFormik.handleChange('email')}
+                error={emailFormik.touched.email && Boolean(emailFormik.errors.email)}
+                errorMessage={emailFormik.touched.email ? emailFormik.errors.email : ''}
+              />
+            </Box>
+          </form>
+        </Box>
+      )}
+
+      {step === Steps.VerifyCode && (
+        <Box width="100%">
+          <Text
+            textAlign="center"
+            variant="h4-semibold"
+          >
+            Verify Email
+          </Text>
+
+          <Text
+            textAlign="center"
+            variant="bs-regular"
+          >
+            We sent you a 6 digit confirmation code to {emailFormik.values.email} Please enter it below to confirm your
+            email address.
+          </Text>
+
+          <Box
+            margin="spacing-md spacing-none"
+            width="100%"
+          >
+            <TextInput
+              placeholder="Enter 6-digit code"
+              value={codeFormik.values.code}
+              onChange={codeFormik.handleChange('code')}
+              error={codeFormik.touched.code && Boolean(codeFormik.errors.code)}
+              errorMessage={codeFormik.touched.code ? codeFormik.errors.code : ''}
+            />
+          </Box>
+
+          <Box
+            display="flex"
+            flexDirection="row"
+            justifyContent="center"
+            gap="spacing-xxxs"
+          >
+            <Text textAlign="center">Didn't receive a code?</Text>
+            <Text
+              textAlign="center"
+              onClick={resendVerificationCode}
+              css={css`
+                text-decoration: underline;
+                cursor: pointer;
+              `}
+            >
+              Send code again
```

Comment:
Some kind of loader should be visible when we click on it or user should not be able to re click  on it.